### PR TITLE
Feature/reports page

### PR DESCRIPTION
**Pull request by Kristute**

### CODE REVIEW

File: components/ArticleDetailPage.vue [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  articleType: string
+}
+const props = defineProps<ArticleDetailProps>()
+
+const article = ref<Article>()
+const route = useRoute()
+const id = computed(() => route.params.id)
+
+onMounted(() => {
+  fetch(
```
Comment (SIL0RAK) [inline components/ArticleDetailPage.vue:59]: what will happen if this request fails?

File: components/ArticleDetailPage.vue [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      id.value
+  )
+    .then((res) => res.json())
+    .then((a) => {
+      article.value = a
+    })
+})
+</script>
+<script lang="ts">
+export default { name: 'PageID' }
```
Comment (SIL0RAK) [inline components/ArticleDetailPage.vue:72]: why do you need this ?

File: src/useArticlesApi.ts [hunk 3]
```diff
@@ -6,7 +6,7 @@ interface ArticlesProps {
   articles: Article[]
 }
 
-export function useArticlesApi() {
+export function useArticlesApi(page) {
```
Comment (SIL0RAK) [inline src/useArticlesApi.ts:9]: I guess typescript got lost at some point :smile:

**Review by SIL0RAK — CHANGES_REQUESTED** [review]

File: components/ArticleDetailPage.vue (hunk 2 above)
Comment (Kristute) [reply components/ArticleDetailPage.vue:72]: Without the name, I was getting an error for `_id.vue` files: 
`Invalid component name: "_id". Component names should conform to valid custom element name in html5 specification.`
But it seems, that when I moved it to `ArticleDetailPage.vue` component, it does not appear again, so I will remove it.

**Review by SIL0RAK — APPROVED** [review]

File: pages/articles/index.vue [hunk 4]
```diff
@@ -17,31 +17,14 @@
...
-  { value: 100, label: '100' },
-  { value: 200, label: '200' },
-]
-
-const { articles, loadArticles, totalArticlesCount, error } = useArticlesApi()
-
-loadArticles(currentAmount, 1)
+const {
+  articles,
+  loadArticles,
```
[the remark this answers is not in the document]
Comment (Kristute) [reply pages/articles/index.vue]: I think this was changed, or I'm missing something?

File: components/ArticleDetailPage.vue [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    </div>
+  </div>
+</template>
+<script setup lang="ts">
+import { ref, Ref } from 'vue'
+import { Article } from '../types/Article'
+interface ArticleDetailProps {
+  articleType: string
+}
+const props = defineProps<ArticleDetailProps>()
```
[the remark this answers is not in the document]
Comment (Kristute) [reply components/ArticleDetailPage.vue:56]: Moved to `useDetailsApi.ts`

**Pull request merged**